### ISSUE TITLE
🐛 BUG:

### ISSUE THREAD

**Issue by LuckyWraptor** [cloudflare/workers-sdk#3922]
### Which Cloudflare product(s) does this pertain to?

D1

### What version(s) of the tool(s) are you using?

3.7.0

### What version of Node are you using?

18.16.0

### What operating system are you using?

Mac Ventura 13.5.2

### Describe the Bug

Wrangler dev will not start with a D1 binding configured in wrangler.toml

```
name = "{REDACTED}"
main = "src/index.js"
compatibility_date = "2023-08-01"

workers_dev = true

[[d1_databases]]
binding = "DBAPI"
database_name = "{REDACTED}"
database_id = "{REDACTED}"

[vars]
APP_NAME = "{REDACTED}"
APP_VERSION = "{REDACTED}"
```

### Please provide any relevant error logs

> export NO_D1_WARNING=true && npx wrangler dev

```bash
 ⛅️ wrangler 3.7.0
------------------
wrangler dev now uses local mode by default, powered by 🔥 Miniflare and 👷 workerd.
To run an edge preview session for your Worker, use wrangler dev --remote
Your worker has access to the following bindings:
- D1 Databases:
  - DBAPI: {REDACTED} ({REDACTED})
- Vars:
  - APP_NAME: "{REDACTED}"
  - APP_VERSION: "{REDACTED}"
⎔ Starting local server...
workerd/server/workerd-api.c++:615: error: wrapped binding module can't be resolved (internal modules only); moduleName = cloudflare-internal:d1-api
workerd/util/symbolizer.c++:98: warning: Not symbolizing stack traces because $LLVM_SYMBOLIZER is not set. To symbolize stack traces, set $LLVM_SYMBOLIZER to the location of the llvm-symbolizer binary. When running tests under bazel, use `--test_env=LLVM_SYMBOLIZER=<path>`.
workerd/jsg/util.c++:275: error: e = workerd/server/workerd-api.c++:635: failed: expected !value.IsEmpty(); global did not produce v8::Value
stack: 101e7fb21 10217bf46 101e39ae4 101e425fb 101e45e40 101e444e5 101e31426 101e30e06 101e30bb4 101e30b9c 103c13007 103c13139 103c11186 101e23753 /usr/lib/dyld@641e; sentryErrorContext = jsgInternalError
service core:user:shuttershop-api: Uncaught Error: internal error
✘ [ERROR] MiniflareCoreError [ERR_RUNTIME_FAILURE]: The Workers runtime failed to start. There is likely additional logging output above.
```

**Comment by LuckyWraptor** [comment]
Disabling the D1 bind in wrangler.toml results int he application working fine

<img width="730" alt="image" src="https://github.com/cloudflare/workers-sdk/assets/38633741/a96fe000-b337-441f-ac62-1d58156ef015">

**Comment by mkpoli** [comment]
I met the same problem on Windows 10, and found the cause was that I didn't export default in my entry point. After I exported default correctly, the problem solved.

**Comment by rozenmd** [comment]
@LuckyWraptor does it work if you move the vars above the d1_databases section?

**Comment by LuckyWraptor** [comment]
> Contributor

My apologies for not updating this issue.
Even completely removing the  database entries still gave me the error.

I've been able to get it up and running, but not sure why it started working.
I do have export default now, so I'm assuming this resolved the issue.

Whenever I get time, I'll try to debug by reverting commits and see when it got fixed.

**Comment by elithrar** [comment]
Is this resolved when using wrangler 3.9.0?

**Comment by mkpoli** [comment]
> Is this resolved when using wrangler 3.9.0?

No, unfortunately this issue still persists in 3.9.0. You can reproduce it by make a new project, add d1_databases and remove `export default app;`. If you either restore the `export default app` or remove d1_databases, the problem is resolved.

**Comment by oligirling** [comment]
I get the same issue with this. It seems it does not work for Service Worker syntax, its only working for ES modules format

**Comment by LuckyWraptor** [comment]
Yep, it seems like it's related to the listening to events rather than using ES modules.

**Comment by penalosa** [comment]
This is a known limitation of D1 that we don't intend to fix. D1 only supports ES module format Workers, while it seems like the Workers mentioned in this thread are using Service Worker syntax. Service Workers are deprecated, and we recommend using [Module Workers](https://developers.cloudflare.com/workers/reference/migrate-to-module-workers/) instead.

To migrate, you can follow the documentation at https://developers.cloudflare.com/workers/reference/migrate-to-module-workers/, which also lists some of the [benefits of moving](https://developers.cloudflare.com/workers/reference/migrate-to-module-workers/) to ES module format.
